corehq/util/context_processors.py

```diff
@@ -57,7 +57,11 @@ def get_per_domain_context(project, request=None):
             allow_report_an_issue = True
         else:
             domain_name = request.couch_user.domain
-            allow_report_an_issue = request.couch_user.has_permission(domain_name, 'report_an_issue')
+            role = request.couch_user.get_domain_membership(domain_name).role
+            if not role and settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS:
+                allow_report_an_issue = False
+            else:
+                allow_report_an_issue = request.couch_user.has_permission(domain_name, 'report_an_issue')
```

Comment:
Wow, I didn't realize we ever used a fallback to `Permission()`.  What would you think of defining an ICDS-specific "no role permission" and referencing that in https://github.com/dimagi/commcare-hq/blob/2b3fff6dd35e9de2e39c7bd9db93f10272e1c24b/corehq/apps/users/models.py#L480-L485

That'd ensure this sort of thing is consistent site-wide.

Reply:
It's a good idea. But, this would be the only thing that we will be overriding in that custom Permissions class. And I doubt if the report-an-issue permission is checked anywhere else.